src/components/buttons/back-button.cjsx

```diff
@@ -18,35 +18,25 @@ BackButton = React.createClass
   contextTypes:
     router: React.PropTypes.func
 
-  getInitialState: ->
-    historyInfo = {}
-    hasHistory = (History.length > 1)
-    # Only gets previous route transitioned to. Routes from router.replaceWith are ignored.
-    # See TransitionStore for more detail.
-    historyInfo = TransitionStore.getPrevious(@context.router.match) if hasHistory
-
-    {hasHistory, historyInfo}
+  navigate: ->
+    historyInfo = TransitionStore.getPrevious(@context.router)
+    if historyInfo.path
+      @context.router.transitionTo(historyInfo.path)
+    else
+      @context.router.transitionTo(@props.fallbackLink.to, @props.fallbackLink.params)
 
   render: ->
-    {hasHistory, historyInfo} = @state
+    # Gets route to last path that was not the same as the current one
+    # See TransitionStore for more detail.
+    historyInfo = TransitionStore.getPrevious(@context.router)
     {fallbackLink, className} = @props
     {text} = fallbackLink
 
-    if hasHistory
-      backText = 'Back'
-      backText = "Back to #{historyInfo.name}" if historyInfo.name
-      backButton = <BS.Button className={className} onClick={@context.router.goBack}>
-        {backText}
-      </BS.Button>
-    else
-      fallbackLink = _.omit(fallbackLink, 'text')
-      classes = 'btn btn-default'
-      classes += " #{className}" if className?
-
-      backButton = <Link {...fallbackLink} className={classes}>
-        {text}
-      </Link>
+    backText = if historyInfo.name then "Back to #{historyInfo.name}"  else fallbackLink.text
 
-    backButton
+    <BS.Button className={className} {...@props}
+      onClick={@navigate}>
```

Comment:
Do you think this could done as a `Router.Link` instead of using `onClick` for `transitionTo`?  I think have links increases usability a little bit, like for weird ppl like me that like to open everything in a new tab.